### DISCUSSION TITLE
Algolia DocSearch Packaging Issue?

### DISCUSSION THREAD

**Discussion by randombeeper** [facebook/docusaurus#10761] [General]
Hello Docusaurus team. I recently saw the following comment in the [Docusaurus Discord channel for DocSearch](https://discord.com/channels/398180168688074762/630841131353112607/1311928655751938149):

> Hi team, in preparation for upgrading to Docusaurus 3.6.3, I deleted my yarn.lock file and re-created it using the dependencies listed in https://github.com/facebook/docusaurus/blob/main/examples/classic/package.json except that I changed 3.6.0 to 3.6.3 for Docusaurus packages. After this, I started getting the following error on build: [ERROR] Client bundle compiled with errors therefore further build is impossible.
> Module not found: Error: Can't resolve 'algoliasearch/dist/algoliasearch-lite.esm.browser' in 'C:\Users\user-name\Repositories\docs-site\components\@docsearch\react\dist\esm' 
> Looking at the newly generated lock file, it looks like algoliasearch 4.24.0 and 5.15.0 are both being installed, with the 4.x being a dependency of docusaurus and the 5.x being a dependency of docsearch.
> In any case, the error went away when I added "@docsearch/react": "3.5.2" as a dependency in packages.json.

I checked with engineering on my side at Algolia and they had this to say:

> Docusaurus likely depends on a version of @docsearch/react that internally references algoliasearch@4.x. Seems like the lock file was pulling both algoliasearch@4.x and algoliasearch@5.x, leading to the conflict. The error would result from @docsearch/react attempting to use algoliasearch@4.x but finding an incompatible version (5.x).
> It's likely @docsearch/react@3.5.2 is the version used in the Docusaurus 3.6.3 setup. If not, this mismatch is likely an oversight.

Would someone on the Docusaurus side be able to confirm this?

**Comment by slorber** [comment]
It looks related to my comment here: https://github.com/facebook/docusaurus/pull/10672#issuecomment-2482664785

cc @millotp
